node/src/components/deploy_buffer.rs

```diff
@@ -422,6 +439,80 @@ impl Drop for DeployBufferMetrics {
     }
 }
 
+mod prune {
+    use super::*;
+
+    /// Prunes a deploy from an individual DeployCollection, returns the total
+    /// deploys pruned
+    pub fn prune_deploys(deploys: &mut DeployCollection, expired: DeployHash) -> usize {
+        let initial_len = deploys.len();
+        deploys.retain(|k, v| k != &expired && !v.dependencies().contains(k));
+        initial_len - deploys.len()
+    }
+    /// Prunes an expired deploy information from each ProtoBlockCollection, returns the total
+    /// deploys pruned
+    pub fn prune_blocks(blocks: &mut ProtoBlockCollection, expired: DeployHash) -> usize {
+        let mut pruned = 0;
+        let mut remove = Vec::new();
+        for (block_hash, deploys) in blocks.iter_mut() {
+            pruned += prune_deploys(deploys, expired);
+            if deploys.is_empty() {
+                remove.push(*block_hash);
+            }
+        }
+        blocks.retain(|k, _v| !remove.contains(&k));
```

Comment:
Since `remove` is a `Vec` and `blocks` is a `HashMap`, wouldn't it be faster to do:
```rust
for block_hash in remove {
    blocks.remove(&block_hash);
}
```